src/server/models/page.js

```diff
@@ -702,6 +702,20 @@ module.exports = function(crowi) {
     return await findListFromBuilderAndViewer(builder, user, false, option);
   };
 
+  /**
+   * find pages that is match with `path` and its descendants whitch user is able to manage
+   */
+  pageSchema.statics.findManageableListWithDescendants = async function(path, user, option = {}) {
+    const builder = new PageQueryBuilder(this.find());
+    builder.addConditionToListWithDescendants(path, option);
+    builder.addConditionToExcludeRedirect();
+
+    // add grant conditions
+    await addConditionToFilteringByViewerToEdit(builder, user);
```

Comment:
以下のコードを含んでしまってもいい気がする。

```
    //  because findManageableListWithDescendants excludes GRANT_RESTRICTED pages
    if (targetPage.grant === GRANT_RESTRICTED) {
      pages.push(targetPage);
    }
```

現時点では revertDeletedPageRecursively だけこのブロックを含まない処理になっているが、含んでも問題にはならないはず。